app/src/main/java/com/jerboa/model/PostViewModel.kt

```diff
@@ -311,4 +315,22 @@ class PostViewModel : ViewModel(), Initializable {
             else -> {}
         }
     }
+
+    fun markPostAsRead(
+        form: MarkPostAsRead,
+        appState: JerboaAppState,
+    ) {
+        appState.coroutineScope.launch {
+            markPostRes = ApiState.Loading
+            markPostRes = apiWrapper(API.getInstance().markAsRead(form))
+
+            when (val markRes = markPostRes) {
+                is ApiState.Success -> {
+                    updatePost(markRes.data.post_view)
+                }
+
+                else -> {}
+            }
+        }
+    }
```

Comment:
The only concern if I have here is, if the if viewmodel gets cleared it wont cancel this coroutine and note sure what this updatePost() will do then, It's why i rather had that `API.getInstance().markAsRead(form)` only this was done in that appState scope. If it is possible can you change that or, will the code below not do anything in that regard then?